src/miner.cpp

```diff
@@ -351,8 +359,12 @@ bool BlockAssembler::isStillDependent(CTxMemPool::txiter iter)
 
 bool BlockAssembler::TestPackageSigOps(uint64_t packageSize, unsigned int packageSigOps)
 {
-    uint64_t nMaxSigOpsAllowed = GetMaxBlockSigOpsCount(nBlockSize + packageSize);
-    if (nBlockSigOps + packageSigOps >= nMaxSigOpsAllowed)
+    if (!may2020Enabled) // if may2020 is enabled, its a constant
+    {
+        maxSigOpsAllowed = GetMaxBlockSigOpsCount(nBlockSize + packageSize);
+    }
+
```

Comment:
don't we need something in here to get the package maxSigOpsAllowed for sigcheck/sigops after the fork as well?

Reply:
prefork, sigops maxSigOpsAllowed is a function of the current block size so we need to calculate it every time we add a tx or package of tx to the block.

postfork, sigchecks maxSigOpsAllowed is a function of the maximum block size so it is a constant throughout the block creation and so is set at the top of BlockAssembler::CreateNewBlock:
    if (may2020Enabled)
    {
        maxSigOpsAllowed = maxSigChecks.Value();
    }

Reply:
ah good